index.d.ts

```diff
@@ -1,6 +1,6 @@
 import * as axios from 'axios';
 
-export = IAxiosRetry;
+export default IAxiosRetry;
```

Comment:
when use `=`, below error occur. So, i change to `default`.

```
error TS1192: Module '"/home/study/workspace/ts-node-oidc/node_modules/axios-retry/lib/esm/index"' has no default export.
```

Reply:
That's the reason for why an import like:

```ts
import { IAxiosRetryConfig } from 'axios-retry'
```

no longer works.

Reply:
@yutak23 how/when did you get that error? Did it work with version of axios-retry prior to your change?

I'm thinking that we could just revert to the old code if your issue was not introduced by those changes.

Reply:
@rchl 

When compiling to CommonJS, `export =` is fine, but when compiling to ES Module, `export =` will result in the following error Therefore, we changed it to default export.
https://github.com/softonic/axios-retry/pull/250#discussion_r1391915371

However, the modification to export `IAxiosRetryConfig` etc. for name import was omitted. Sorry about that. We have uploaded the fix in the https://github.com/softonic/axios-retry/pull/252.

Reply:
Did `export =` really create an issue and in what circumstances? I haven't seen issues after changing it in this project and building.

Reply:
> I haven't seen issues after changing it in this project and building.

I would be very grateful if you could provide me with details of the configuration information for your TypeScript project.
The problem I am talking about happens when compiling to ES Module with ES Module configuration (set `type: module` in package.json)

When `export =`, the following compile error occurs. When `export default` is used, no error occurs.
![image](https://github.com/softonic/axios-retry/assets/79501292/f4adf139-886c-469f-8ca2-7a1ccc9d06e5)
![image](https://github.com/softonic/axios-retry/assets/79501292/7574778d-1be3-4ab6-94a5-b17495367e86)

My environment is below.

```json
// package.json
	"type": "module",
	"dependencies": {
		"axios": "^1.6.2",
		"axios-retry": "^3.9.0",
	},
	"devDependencies": {
		"@tsconfig/node16": "^16.1.1",
		"@tsconfig/strictest": "^2.0.2",
		"typescript": "~5.2.2",
	}
```
```json
// tsconfig.json
{
	"extends": ["@tsconfig/node16/tsconfig.json", "@tsconfig/strictest/tsconfig.json"],
	"compilerOptions": {
		"outDir": "dist",
		"declaration": true,
		"sourceMap": true
	},
	"include": ["srv/**/*.ts"],
	"exclude": ["node_modules", "dist"]
}
```

Reply:
My question is: did this work before (before v3.9.0). Because if not then there is not really a need to change that now. Could do that in next major version if needed.

Reply:
I know this is already becoming Revert's policy, but I think it would be rude not to answer your question, so I want to answer.

The type definition used to be `export default` and it worked except for the ES Module. . However, this had a problem with [issue](https://github.com/softonic/axios-retry/issues/228) and has been modified. But then another problem arose and more changes were made.
https://github.com/softonic/axios-retry/blob/0f50dc99075ff27d02012d0c993e52ef45b6ec6c/index.d.ts#L66